lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -720,15 +732,23 @@ def delete_batch(
         errors: List[DatasetErrorMessage] = []
         for dataset in payload.datasets:
             try:
-                manager = self.dataset_manager_by_type[dataset.src]
-                dataset_instance = manager.get_owned(dataset.id, trans.user)
-                manager.error_unless_mutable(dataset_instance.history)
-                if dataset.src == DatasetSourceType.hda:
-                    self.hda_manager.error_if_uploading(dataset_instance)
-                if payload.purge:
-                    manager.purge(dataset_instance, flush=True)
+                if dataset.src == "hda":
+                    hda_manager = self.dataset_manager_by_type[dataset.src]
+                    hda = hda_manager.get_owned(dataset.id, trans.user)
+                    hda_manager.error_if_uploading(hda)
+                    hda_manager.error_unless_mutable(hda.history)
```

Comment:
I wonder if one more dictionary called `raise_delete_error_by_type` that will call these 2 methods for `hda`s and just no-op for `ldda`s will eliminate this duplication :thinking: